File: app/src/main/java/party/liyin/mipushfakeframework/MainActivity.java
package party.liyin.mipushfakeframework;

import android.os.Bundle;
import android.app.Activity;
import android.text.Html;
import android.text.method.LinkMovementMethod;
import android.widget.TextView;

import com.xiaomi.xmsf.R;

public class MainActivity extends Activity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);
        //noinspection deprecation
//        TextView descrView = ((TextView)findViewById(R.id.decrView));
//        descrView.setText(Html.fromHtml(getString(R.string.descr)));
//        descrView.setMovementMethod(LinkMovementMethod.getInstance());
    }

}

File: app/src/main/java/party/liyin/mipushfakeframework/MainActivity.kt
package party.liyin.mipushfakeframework

import android.os.Bundle
import android.app.Activity
import kotlinx.android.synthetic.main.activity_main.*

import com.xiaomi.xmsf.R
import party.liyin.mipushfakeframework.utils.FakeBuildUtils

class MainActivity : Activity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        fakebuild_active.setOnClickListener { runFake() }
        check()
    }

    fun check(){
        Thread(Runnable {
            val isMiuiBuild = FakeBuildUtils.isMiuiBuild()
            runOnUiThread { fakebuild.setText(getString(R.string.fake_build_status,getString(if(isMiuiBuild) R.string.fake_boot_success else R.string.fake_boot_fail))) }
        }).start()
    }

    fun runFake() {
        Thread(Runnable {
            val isMiuiBuild = FakeBuildUtils.insertMiui()
            runOnUiThread { fakebuild.setText(getString(R.string.fake_build_status,getString(if(isMiuiBuild) R.string.fake_boot_success else R.string.fake_boot_fail))) }
        }).start()
    }
}
